#--------------------------------------------------------------------------------------------
# General Parameters. Adjust them!
#--------------------------------------------------------------------------------------------
general:
  data_path: /home/daybeha/Documents/dataset/kitti/
  sequence: 00

#--------------------------------------------------------------------------------------------
# Camera Parameters. Adjust them!
#--------------------------------------------------------------------------------------------
camera:
  # Color order of the images (0: BGR, 1: RGB. It is ignored if images are grayscale)
  RGB: 1

  # Camera resolution
  width: 1241
  height: 376

## 00 -02
#  fx: 7.188560000000e+02
#  fy: 7.188560000000e+02
#  cx: 6.071928000000e+02
#  cy: 1.852157000000e+02
#   04 -12
  fx: 7.070912000000e+02
  fy: 7.070912000000e+02
  cx: 6.018873000000e+02
  cy: 1.831104000000e+02


ratslam:
  image_crop_x_min: 100
  image_crop_x_max: 1150 # 1100
  image_crop_y_min: 50
  image_crop_y_max: 300
  down_sample: 2
  vt_match_threshold :  0.085
  template_x_size: 60
  template_y_size: 10
  vt_shift_match :  5
  vt_step_match :  1
  vt_normalisation :  0.4
  pc_vt_restore: 0.04
  pc_cell_x_size :  2
  pc_dim_xy :  30
  exp_delta_pc_threshold :  1.0 # 两个odom之间的位置变化最大值，超过该值要寻找回环。数据集的速度越快，这个值应当越大
  pc_vt_inject_energy :  0.15
  vt_active_decay :  1.0

  exp_loops: 20

  act_level_threshold: 90 # 90 for urbans, 100 for kitti
  min_delta_threshold: 2.0  # 2.0,  8.0 for kitti02
  scale_factor: 10
  vocab_file: /home/daybeha/Documents/github/Hybird-NeuroSLAM/myratslam_ws/src/ratslam_ros/src/gaussianslam/orbvoc.dbow3
  min_loop_frame_dis: 100
  loop_cnt_thres: 2  # 1 for keyframe, 2 for normal
  loop_score_thres: 0.045 # 0.045 for kitti, 0.05 for urbans

  HD_distribution: 1  # 0: Bayesian; 1: Von Mises
  cali_mode: 0 #0:直接拉回原细胞位置 1:校准方向 2:校准方向和位置

draw:
  enable: true
  vt_window_width: 640
  vt_window_height: 900
  exp_map_size: 500
  posecells_size: 250
  media_path: /home/daybeha/Documents/github/Hybird-NeuroSLAM/myratslam_ws/src/ratslam_ros/src/media
  image_file: irat_sm.tga